pipelines/build/common/build_base_file.groovy

```diff
@@ -318,13 +319,21 @@ class Builder implements Serializable {
         if (matcher.matches()) {
             return Integer.parseInt(matcher.group('version'))
         } else if ("jdk".equalsIgnoreCase(javaToBuild.trim())) {
-            // Query the Adopt api to get the "tip_version"
-            def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
-            context.println "Querying Adopt Api for the JDK-Head number (tip_version)..."
-
-            def response = JobHelper.getAvailableReleases(context)
-            int headVersion = (int) response.getAt("tip_version")
-            context.println "Found Java Version Number: ${headVersion}"
+            int headVersion
+            try {
+                context.timeout(time: 1, unit: "HOURS") {
+                    // Query the Adopt api to get the "tip_version"
+                    def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
+                    context.println "Querying Adopt Api for the JDK-Head number (tip_version)..."
+
+                    def response = JobHelper.getAvailableReleases(context)
+                    headVersion = (int) response.getAt("tip_version")
+                    context.println "Found Java Version Number: ${headVersion}"
+                }
+            } catch (FlowInterruptedException e) {
+                context.println "[ERROR] Adopt API Request timeout (1 HOUR) has been reached. Exiting..."
```

Comment:
I'd parameterise these error messages with the timeout logic above (as opposed to writing in a string)

Reply:
That's the plan! I'm still working on getting the gradle dependancies sorted out first, then I'm going to look into the output messages